src/CodeCoverageExtension.php

```diff
@@ -50,17 +49,13 @@ public function load(ServiceContainer $container, array $params = []): void
         });
 
         $container->define('code_coverage', static function ($container) {
-            try {
-                $coverage = new CodeCoverage(null, $container->get('code_coverage.filter'));
-            } catch (RuntimeException $error) {
-                throw new NoCoverageDriverAvailableException(
-                    'There is no available coverage driver to be used.',
-                    0,
-                    $error
-                );
+            if (version_compare(Version::id(), '9.0.0') >= 0) {
+                $driver = (new Selector())->forLineCoverage($container->get('code_coverage.filter'));
+            } else {
+                $driver = null;
             }
 
-            return $coverage;
+            return new CodeCoverage($driver, $container->get('code_coverage.filter'));
```

Comment:
PHPStan complains, because it doesn't get the version check... I'm not sure how to work around this. @drupol Any ideas?

Reply:
I'll try to give it a try soon!

Reply:
Here's my [log](https://del.dog/icupelivyr.go)... investigating.

Reply:
Ok I see.

CodeCoverage's first parameter is not supposed to be null, but in the code you wrote, it is possible that `$driver` gets `null`.

How about doing this?

```php
        $container->define('code_coverage', static function ($container) {
            if (version_compare(Version::id(), '9.0.0') < 0) {
                throw new Exception(
                    'There is no available coverage driver to be used.'
                );
            }

            /** @var Filter $filter */
            $filter = $container->get('code_coverage.filter');

            return new CodeCoverage(
                (new Selector())->forLineCoverage($container->get('code_coverage.filter')),
                $filter
            );
        });
```

Reply:
Hum, with this code, the coverage will not be possible in version under 9.0.0. I think that we need to maintain the `null` value.

I'll go for something like this : 

```php
        $container->define('code_coverage', static function ($container) {
            /** @var Filter $filter */
            $filter = $container->get('code_coverage.filter');

            if (version_compare(Version::id(), '9.0.0') < 0) {
                // This is valid in version under 9 and let the package choose the correct tool
                return new CodeCoverage(null, $filter);
            } else {
                return new CodeCoverage(
                    (new Selector())->forLineCoverage($container->get('code_coverage.filter')),
                    $filter
                );
        });
```

Also, I think that catching the exception and throwing our own is better from a DX perspective because it'll help knowing where the issue comme from…

Reply:
Oh... if this is valid, then this is indeed the proper way to go. We could also eventually get rid of the else clause.

```php
$container->define('code_coverage', static function ($container) {
            /** @var Filter $filter */
            $filter = $container->get('code_coverage.filter');

            if (version_compare(Version::id(), '9.0.0') < 0) {
                // This is valid in version under 9 and let the package choose the correct tool
                return new CodeCoverage(null, $filter);
            }

                return new CodeCoverage(
                    (new Selector())->forLineCoverage($container->get('code_coverage.filter')),
                    $filter
                );
        });
```

Reply:
I added this, but even with `// @phpstan-ignore-line` I'm getting errors...